labgrid/driver/power/netio_kshell.py

```diff
@@ -1,37 +1,46 @@
 """ tested with NETIO 4C, should be compatible with all NETIO 4-models """
 
 import re
-import telnetlib
+
+import pexpect
 
 PORT = 1234
 
 def power_set(host, port, index, value):
     index = int(index)
     assert 1 <= index <= 4
     value = "1" if value else "0"
-    tn = telnetlib.Telnet(host, port, 1)
-    tn.read_until(b"\r\n", 0.5)
-    tn.write(b"login admin admin\n")
-    tn.read_until(b"250 OK\r\n", 0.5)
-    tn.write(f"port {index} {value}\n".encode())
-    tn.read_until(b"250 OK\r\n", 0.5)
-    tn.write(b"quit\n")
-    tn.close()
+
+    with pexpect.spawn(f"telnet {host} {port}", timeout=1) as tn:
+        tn.expect(b"100 HELLO .*\r\n")
```

Comment:
This wasn't in the old version. Where did you find it?

Reply:
In the manual linked above. If that does not apply to all versions, we need something else. As I don't have a NetIO device, I have no idea what to use else.